lib/improver/utilities/cube_manipulation.py

```diff
@@ -157,51 +152,435 @@ def _strip_var_names(cubes):
+    """
+    unmatching_attributes = compare_attributes(cubes)
+    if len(unmatching_attributes) > 0:
+        for i, cube in enumerate(cubes):
+            # Remove history.
+            if "history" in unmatching_attributes[i]:
+                cube.attributes.pop("history")
+                unmatching_attributes[i].pop("history")
+            # Normalise grid_id to ukx_standard_1
+            if "grid_id" in unmatching_attributes[i]:
+                if cube.attributes['grid_id'] in ['enukx_standard_v1',
+                                                  'ukvx_standard_v1',
+                                                  'ukx_standard_v1']:
+                    cube.attributes['grid_id'] = 'ukx_standard_v1'
+                unmatching_attributes[i].pop("grid_id")
+            # Add model_id if titles do not match.
+            if "title" in unmatching_attributes[i]:
+                model_title = cube.attributes.pop('title')
+                new_model_id_coord = build_coordinate([1000*i],
+                                                      long_name='model_id',
+                                                      data_type=np.int)
+                new_model_coord = build_coordinate([model_title],
+                                                   long_name='model',
+                                                   coord_type=AuxCoord,
+                                                   data_type=np.str)
+                cube.add_aux_coord(new_model_id_coord)
+                cube.add_aux_coord(new_model_coord)
+                unmatching_attributes[i].pop("title")
+            # Remove any other mismatching attributes but raise warning.
+            if len(unmatching_attributes[i]) != 0:
+                for key in unmatching_attributes[i]:
+                    msg = ('Do not know what to do with ' + key +
+                           ' will delete it'
+                           ' - value is {}'.format(cube.attributes[key]))
+                    warnings.warn(msg)
+                    cube.attributes.pop(key)
+    return cubes
+
+
+def _equalise_cube_coords(cubes):
+    """
+    Function to equalise coordinates that do not match.
+
+    Args:
+        cubes : Iris cubelist
+            List of cubes to check the coords and revise.
+    Returns:
+        cubelist : Iris cubelist
+            List of cubes with revised coords.
+            The number of cubes in cubelist
+            may be greater than the original number
+            of cubes as the original cubes will be sliced
+            so that that they can be merged together.
+            Merging can only create new coords not add
+            to existing mismatching coords.
+            Note: This internal function modifies the incoming cubes
+    Raises:
+        ValueError: If coordinates in error_keys do not match.
+        ValueError: If model_id has more than one point.
+    """
+    unmatching_coords = compare_coords(cubes)
```

Comment:
Could you add a comment to indicate that you expect the number of cubes and number of `unmatching_coords` to be the same?

Reply:
Done